torchsynth/module.py

```diff
@@ -227,7 +227,65 @@ def update_device(self, device=None):
             parameter.parameter_range.to(device)
 
 
-class ADSR(SynthModule):
+class ControlRateModule(SynthModule):
+    """
+    Control rate module is an abstract base class for SynthModules that run
+    at a control rate as opposed to full audio rate.
+
+    Optionally, the output control signal can be upsampled to the correct
+    sample rate and buffer size.
+    """
+
+    def __init__(
+        self,
+        synthglobals: SynthGlobals,
+        upsample: Optional[bool] = False,
```

Comment:
Can you document when upsample should and should not be true? with maybe an example when you wouldn't want to?

Reply:
Yes. Also like the idea of it not being a default